README.md

```diff
@@ -76,15 +76,25 @@ API
 
 The public Accumulo API is composed of :
 
- * All public classes and interfaces in the org.apache.accumulo.core.client
-   package, as as well as all of its subpackages excluding those named *impl*.
- * Key, Mutation, Value, Range, Condition, and ConditionalMutation in
-   org.apache.accumulo.core.data.
- * All public classes and interfaces in the org.apache.accumulo.minicluster
-   package, as well as all of its subpackages excluding those named *impl*.
- * Anything with public or protected acccess within any Class or Interface that
-   is in the public API. This includes, but is not limited to: methods, members
-   classes, interfaces, and enums.
+All public types in the following packages and their subpackages excluding
+those named *impl*, *thrift*, or *crypto*. 
+
+   * org.apache.accumulo.core.client
+   * org.apache.accumulo.core.data
+   * org.apache.accumulo.core.security
+   * org.apache.accumulo.minicluster
+
+A type is a class, interface, or enum.  Anything with public or protected
```

Comment:
>  The only potential drawback I can think of is making the stmt so long that no one reads its

I think this is one time where it benefits us to be as specific as possible (see how often we had these "oops" moments already, and that Sean and I didn't know which was correct).
> I have heard @ctubbsii mention that we build sealed jars that would prevent users from doing this.

Sealed jars are only relevant if the user downloads binary artifacts from us, not from users building the code themselves, no?